From 7c9c080333d5340c2c031ad4258489fe3f91b64a Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sun, 20 Feb 2022 15:42:27 +0100
Subject: [PATCH 1/4] forms/fields compatibility with django >= 4.0

src/postorius/forms/fields.py:
Use `django.utils.encoding.smart_str` (introduced with django 4.0) by
default and fall back to `django.utils.encoding.smart_text` (django <
4.0) otherwise.
---
 src/postorius/forms/fields.py | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/postorius/forms/fields.py b/src/postorius/forms/fields.py
index f1578615..6235260b 100644
--- a/src/postorius/forms/fields.py
+++ b/src/postorius/forms/fields.py
@@ -19,7 +19,14 @@
 
 
 from django import forms
-from django.utils.encoding import smart_text
+
+
+try:
+    from django.utils.encoding import smart_str
+except ImportError:
+    # django < 4.0
+    from django.utils.encoding import smart_text as smart_str
+
 from django.utils.translation import gettext_lazy as _
 
 from postorius.utils import with_empty_choice
@@ -62,7 +69,7 @@ class ListOfStringsField(forms.Field):
             line = line.strip()
             if not line:
                 continue
-            result.append(smart_text(line))
+            result.append(smart_str(line))
         return result
 
 
-- 
GitLab


From 1b711c0adb063f165e00ed604a398c769f34b9e3 Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sun, 20 Feb 2022 15:45:19 +0100
Subject: [PATCH 2/4] Move the upper boundary for django to < 4.1

setup.py:
As postorius is compatible with django 4.0, set the upper boundary to <
4.1.
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 9a900eee..18374c10 100644
--- a/setup.py
+++ b/setup.py
@@ -54,7 +54,7 @@ setup(
     include_package_data=True,
     install_requires=[
         'django-mailman3>=1.3.7',
-        'django>=2.2,<3.3',
+        'django>=2.2,<4.1',
         'mailmanclient>=3.3.3',
         'readme_renderer[md]',
     ],
-- 
GitLab


From 3dfa2cca5ea7d764e69e6a4c37feb3808755b69d Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sun, 20 Feb 2022 15:46:30 +0100
Subject: [PATCH 3/4] Add python 3.10 and django 4.0 support to tox

tox.ini:
Add environments for python 3.10 and django 4.0
---
 tox.ini | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tox.ini b/tox.ini
index 744dbbc8..bce92d5a 100644
--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist = py{36,37,38,39}-django{22,31,32,-latest},qa
+envlist = py{36,37,38,39,310}-django{22,31,32,-latest},qa
 
 [testenv]
 usedevelop = True
@@ -20,6 +20,7 @@ deps =
     django30: Django>=3.0,<3.1
     django31: Django>=3.1,<3.2
     django32: Django>=3.2,<3.3
+    django40: Django>=4.0,<4.1
     django-latest: https://github.com/django/django/archive/main.tar.gz
 commands =
   pytest {posargs:../src}
@@ -31,7 +32,7 @@ setenv =
 basepython = python3
 deps =
     {[testenv]deps}
-    Django>=3.2,<3.3
+    Django>=3.2,<4.1
     coverage
 commands =
     coverage run -m pytest {posargs:../src/postorius}
-- 
GitLab


From b5e9c49cf710295f53b340f22c7d2ec5cbb7a9da Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Sun, 20 Feb 2022 15:47:44 +0100
Subject: [PATCH 4/4] Add python 3.10 and django 4.0 targets to gitlab CI

.gitlab-ci.yml:
Run tests against latest django using python 3.10.
Add test targets against django 3.2 and 4.0 on python 3.10.
---
 .gitlab-ci.yml | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/.gitlab-ci.yml b/.gitlab-ci.yml
index ba6cea8c..9e9da7f4 100644
--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -66,13 +66,23 @@ py39-django32:
   script:
     - tox -e py39-django32
 
-py39-django-latest:
+py310-django32:
+  stage: test
+  script:
+    - tox -e py310-django32
+
+py310-django40:
+  stage: test
+  script:
+    - tox -e py310-django40
+
+py310-django-latest:
   stage: test
   allow_failure: true
   script:
     - printf "Remove all django version restriction from setup.py.\n"
     - sed -E 's/django[<=>,.[:digit:]]+/django/' -i setup.py
-    - tox -e py39-django-latest
+    - tox -e py310-django-latest
 
 coverage:
   stage: test
-- 
GitLab

